proposals/tests/proposal_submission_tests.py

```diff
@@ -0,0 +1,241 @@
+from django.contrib.auth.models import User, Group, AnonymousUser
+from django.conf import settings
+from django.test import TestCase
+from django.core.management import call_command
+
+from main.tests import BaseViewTestCase
+from proposals.models import Proposal, Relation
+from reviews.models import Review
+
+from proposals.views.proposal_views import (
+    ProposalSubmit,
+    ProposalSubmitPreApproved,
+    ProposalSubmitPreAssessment,
+)
+
+
+class BaseProposalTestCase(TestCase):
+    fixtures = [
+        "groups",
+        "settings",
+        "registrations",
+        "fundings",
+        "institutions",
+        "relations",
+        "studentcontexts",
+        "agegroups",
+        "compensations",
+        "recruitments",
+        "specialdetails",
+        "traits",
+        "registrationkinds",
+        "registrations",
+    ]
+
+    def setUp(self):
+        self.setup_users()
+        self.setup_proposal()
+
+    def setup_users(self):
+        self.secretary = User.objects.create_user(
+            "secretary",
+            "test@test.com",
+            "secret",
+            first_name="The",
+            last_name="Secretary",
+        )
+        self.c1 = User.objects.create_user("c1", "test@test.com", "secret")
+        self.c2 = User.objects.create_user("c2", "test@test.com", "secret")
+        self.user = User.objects.create_user(
+            "user", "test@test.com", "secret", first_name="John", last_name="Doe"
+        )
+        self.supervisor = User.objects.create_user(
+            "supervisor", "test@test.com", "secret", first_name="Jane", last_name="Roe"
+        )
+
+        self.secretary.groups.add(
+            Group.objects.get(
+                name=settings.GROUP_PRIMARY_SECRETARY
+            )
+        )
+        self.secretary.groups.add(
+            Group.objects.get(
+                name=settings.GROUP_SECRETARY
+            )
+        )
+
+        self.c1.groups.add(
+            Group.objects.get(name=settings.GROUP_LINGUISTICS_CHAMBER)
+        )
+        self.c2.groups.add(
+            Group.objects.get(name=settings.GROUP_LINGUISTICS_CHAMBER)
+        )
+
+    def setup_proposal(self):
+        """
+        Load our test proposals from a fixture, then add our user as
+        an applicant to each of them.
+        """
+        call_command("loaddata", "proposals/tests/test_proposals.json")
```

Comment:
AFAIK this requires the fixture to exist in the `app/fixtures` directory. For most of our current django apps, it's common to load all fixtures in those directories on first setup.

Putting this test fixture in there would break that process, so I decided not to. My logic: the fixtures dirs contain "need to load" data for correct functioning, and incidental/testing fixtures belong elsewhere.

Reply:
Please don't take this as a 'that's wrong and you're wrong'; it's a 'ehhh, we're misusing fixtures so maybe we should think of something'. I'm just as guilty. 

Officially, fixtures exist only for test data. (That's why `TestCase` is the only place that actually has an API for loading it, aside from management commands) So, the fact that we're using it for initial 'need to load' data is technically a wrong use of fixtures. 

So your reasoning is actually completely the other way around from what it should be... 

However, even if it's a 'wrong' use of fixtures, I do think it's a valid use-case. So... I'd suggest the following:
- Move the test fixture to the fixtures dir, maybe as a subdir to keep some seperation
- Implement* a management command to load all fixtures needed for proper functioning in one go

\* With implement I mean copy over that exact command from DIAPP, where we ran into this issue as well. (And turns out, having a dedicated command for that is actually kinda nice. No complicated bash scripts or forgetting fixtures)

Reply:
I legitimately thought that we used fixtures exactly as intended. 

You make a great suggestion. I have copied over the `load_fixtures` command from DIAPP (thanks!) and have housed testing fixtures in a subdir of the regular fixtures dir. Please note that I made a small change to the command, it iterates over lists rather than loading single path strings.

Considering we do still use fixtures for initial data for the time being, I'm keeping them in separate dirs, with the `load_fixtures` command not loading anything in the `fixtures/testing` subdir.